src/features/ui/Member/Member.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import { AsyncText } from '@zero-tech/zui/lib/types';
+import { getEtherscanWalletUrl } from '@zero-tech/zapp-utils/web3/etherscan';
+import { truncateAddress } from '../../../lib/util/domains/domains';
+import { useWeb3 } from '../../../lib/hooks/useWeb3';
+
+import { SkeletonText } from '@zero-tech/zui/components';
+
+import styles from './Member.module.scss';
+
+export interface MemberProps {
+	title: string;
+	walletAddress: string | AsyncText;
+}
+
+export const Member = ({ title, walletAddress }: MemberProps) => {
+	const { chainId } = useWeb3();
+
+	const isAsyncText = typeof walletAddress === 'object';
+	const address = isAsyncText ? walletAddress.text : walletAddress;
+	const url = address ? getEtherscanWalletUrl(address, chainId) : undefined;
+
+	return (
+		<div className={styles.Member}>
+			<label className={styles.Title}>{title}</label>
+			<Link href={url}>
+				<Address walletAddress={walletAddress} />
+			</Link>
+		</div>
+	);
+};
+
+/*************************
+ * Link
+ *************************/
+
+interface LinkProps {
+	href?: string;
+	children: React.ReactNode;
+}
+
+const Link = ({ href, children }: LinkProps) => {
+	if (href) {
+		return (
+			<a
+				href={href}
+				target={'_blank'}
+				rel={'noreferrer'}
+				className={styles.Address}
+			>
+				{children}
+			</a>
+		);
+	} else {
+		return <span className={styles.Title}>{children}</span>;
+	}
+};
+
+/*************************
+ * Address
+ *************************/
+
+interface AddressProps {
+	walletAddress: MemberProps['walletAddress'];
+}
+
+const Address = ({ walletAddress }: AddressProps) => {
```

Comment:
quick note : Not for this PR but I think we should probs add `Address` to zUI. I think a similar component is being used in `HistoryItem` and imagine shall be used elsewhere

Reply:
Sounds good - I'll make a task for it.